common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/requests/resetpassword/ResetPasswordStartRequest.kt

```diff
@@ -0,0 +1,83 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.providers.nativeauth.requests.resetpassword
+
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.common.java.providers.nativeauth.requests.NativeAuthRequest
+import com.microsoft.identity.common.java.util.ArgUtils
+import java.net.URL
+
+/**
+ * Represents a request to the Reset Password /start endpoint, and provides a create() function to instantiate the request using the provided parameters.
+ */
+data class ResetPasswordStartRequest private constructor(
+    override var requestUrl: URL,
+    override var headers: Map<String, String?>,
+    override var parameters: NativeAuthRequestResetPasswordStartParameters
+) : NativeAuthRequest() {
+
+    companion object {
+        /**
+         * Returns a request object using the provided parameters.
+         * The request URL and headers passed will be set directly.
+         * The clientId, password reset token, and challengeType will be mapped to the NativeAuthRequestResetPasswordStartParameters object.
+         *
+         * Parameters that are null or empty will throw a ClientException.
+         * @see com.microsoft.identity.common.java.exception.ClientException
+         */
+        fun create(
+            clientId: String,
+            username: String,
+            challengeType: String,
+            requestUrl: String,
+            headers: Map<String, String?>
+        ): ResetPasswordStartRequest {
+            // Check for empty Strings and empty Maps
+            ArgUtils.validateNonNullArg(clientId, "clientId")
+            ArgUtils.validateNonNullArg(username, "username")
+            ArgUtils.validateNonNullArg(challengeType, "challengeType")
+            ArgUtils.validateNonNullArg(requestUrl, "requestUrl")
+            ArgUtils.validateNonNullArg(headers, "headers")
+
+            return ResetPasswordStartRequest(
+                requestUrl = URL(requestUrl),
+                headers = headers,
+                parameters = NativeAuthRequestResetPasswordStartParameters(
+                    clientId = clientId,
+                    username = username,
+                    challengeType = challengeType
+                )
+            )
+        }
+    }
+
+    /**
+     * NativeAuthRequestResetPasswordStartParameters represents the request parameters sent as part of
+     * /resetpassword/start API call
+     */
+    data class NativeAuthRequestResetPasswordStartParameters(
+        val username: String,
+        @SerializedName("client_id") override val clientId: String,
```

Comment:
Why override here? and what does it do?

I have this question for all the places. some params have override and some don't...why and why not? and what difference does it make?

Reply:
`override` keyword is used where the attribute is also present in the base class.